source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,831 @@
+        $conn->.... // do some query
+    }
+```
+
+## Plugin Install / Update
+
+During plugin installation / deinstallation / update / activate / deactivate a method on the plugin bootstrap is called that can optionally be overwritten. You can do a lot of things with the provided context, e.g.:
+
+- stop process by throwing an exception and notify user with a message
+- notify user on success with a message
+- flush specified caches
+- within update(), addtionally: get currently installed version number of your plugin
+- wihtin secureUninstall() keep user generated data, if he wishes so
+
+Checkout the examples:
+
+```php
+<?php
+
+class Shopware_Plugins_Frontend_SwagSloganOfTheDay_Bootstrap extends Shopware_Components_Plugin_Bootstrap
+{
+    public function afterInit()
+    {
+        $this->get('Loader')->registerNamespace(
+            'ShopwarePlugins\PluginName',
+            $this->Path()
+        );
+    }
+    
+    public function install()
+    {
+        
+    }
+    
+    public function uninstall()
+    {
+            
+    }
+
+    public function secureUninstall()
+    {
+        
+    }
+
+    public function update()
+    {
+        // Check if Shopware version matches
+        if (!$this->assertMinimumVersion('5.2.0')) {
+            throw new Exception('This plugin requires Shopware 5.2.0 or a later version');
+        }
+               
+    }
+    
+    public function enable()
+    {
+        return ['success' => true, 'invalidateCache' => ['proxy', 'frontend', 'backend', 'theme']];
+    }
+    
+    public function disable()
+    {
+        return ['success' => true, 'invalidateCache' => ['proxy', 'frontend', 'backend', 'theme']];
+    }
+
+}
+```
+
+### Decorate a service
+The following example shows you how to decorate a service which implements an interface and gets defined in the Shopware dependency injection container.
+```php
+<?php
+
+namespace SwagExample\Bundle\StoreFrontBundle;
+
+use Shopware\Bundle\StoreFrontBundle\Service\ListProductServiceInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\ProductContextInterface;
+
+class ListProductServiceDecorator implements ListProductServiceInterface
+{
+    private $service;
+
+    public function __construct(ListProductServiceInterface $service)
+    {
+        $this->service = $service;
+    }
+
+    public function getList(array $numbers, ProductContextInterface $context)
+    {
+        $products = $this->service->getList($numbers, $context);
+        //...
+        return $products;
+    }
+
+    public function get($number, ProductContextInterface $context)
+    {
+        return array_shift($this->getList([$number], $context));
+    }
+}
+```
+The original `\Shopware\Bundle\StoreFrontBundle\Service\Core\ListProductService` defined with the service id `shopware_storefront.list_product_service`. The following service definition decorates this service using the service above:
+
+The after init resource event is quite important in this case.
```

Comment:
:D lol, you should show an example! how is the after init resource event called, how about the naming of this event, etc. sorry you mistaken me here :-)

Reply:
D